### ISSUE TITLE
max problem reports reduced cost with wrong sign

### ISSUE THREAD

**Issue by svigerske** [coin-or/Cbc#125]
Issue created by migration from Trac.

Original creator: guest

Original creation time: 2012-12-12 05:53:39

Assignee: @tkralphs

Keywords: reduced cost, maximization, wrong sign, max

When solving a MAX problem and a variable is pushed to its LOWER bound cbc is reporting a positive value for the reduced cost. I believe in practice it should be reported as negative. For the following three complementary cases, cbc seems to report the correct sign for the reduced cost

max problem -> var at UB -> (+) r.c.
min  problem -> var at LB ->  (+) r.c.
min  problem -> var at UB -> (-) r.c.

The problematic case (which occurs when using the LP file interface) can be tested using the following simple LP file:

```
max
obj:
-1.000000 x(1)

s.t.

bounds
   -1 <= x(1) <= 1
end
```

This happens in 2.7, 2.8, and trunk. Below is my contact info:

name: Gabe
email: gabehack`@`gmail.com

**Comment by svigerske** [comment]
Comment by guest created at 2012-12-12 05:56:20

I don't like how some of that formatting came out. See below for what are the working cases...

max problem -> var at UB -> (+) r.c.

min problem -> var at LB -> (+) r.c.

min problem -> var at UB -> (-) r.c.

**Comment by svigerske** [comment]
Probably related to the LP-file reader converting the maximization into a minimization problem.
@jjhforrest Do one actually need to convert to minimization? Could Cbc (or Clp) flip the duals back when reporting a solution?

What seems to be expected is that the last number in the following is a -1 instead of a 1:
```
$ ./bin/cbc  issue125.lp -solve -solu issue125.solu && cat issue125.solu 
Welcome to the CBC MILP Solver 
Version: Trunk (unstable) 
Build Date: Mar 12 2019 
Revision Number: 2526 

command line - ./bin/cbc issue125.lp -solve -solu issue125.solu (default strategy 1)
 CoinLpIO::readLp(): Maximization problem reformulated as minimization
Empty problem - 0 rows, 1 columns and 0 elements
Optimal - objective value -1
Optimal objective -1 - 0 iterations time 0.002
Total time (CPU seconds):       0.00   (Wallclock seconds):       0.00

Optimal - objective value -1.00000000
      0 x(1)                  -1                       1
```

**Comment by jjhforrest** [comment]
Changed so Cbc/Clp flip back to maximize

**Comment by svigerske** [comment]
The change from https://github.com/coin-or/Clp/commit/680189c9d5555bdc2f49a325af38c8f2d2ca1c01 seems to fix this for Clp, but the corresponding change in Cbc seems to have been lost.

**Comment by svigerske** [comment]
Fixed for Cbc by coin-or/Clp@1041a9e551b
```
Welcome to the CBC MILP Solver 
Version: Trunk (unstable) 
Build Date: Apr  6 2019 
Revision Number: 2526 

command line - ./bin/cbc issue125.lp -solve -solu issue125.solu (default strategy 1)
 CoinLpIO::readLp(): Maximization problem reformulated as minimization
Coin0009I Switching back to maximization to get correct duals etc
Empty problem - 0 rows, 1 columns and 0 elements
Optimal - objective value 1
Optimal objective 1 - 0 iterations time 0.002
Total time (CPU seconds):       0.00   (Wallclock seconds):       0.00

Optimal - objective value 1.00000000
      0 x(1)                  -1                      -1
```
Thanks.

**Comment by ghackebeil** [comment]
thanks!
